src/components/DisplayArticlesData.jsx

```diff
@@ -1,60 +1,43 @@
-import React, { useState, useEffect } from 'react';
-import { getArticleData } from '../modules/articlesData';
+import React, { useEffect } from 'react';
+import { getArticleData, getSpecificArticle } from '../modules/articlesData';
+import ArticleCard from './ArticleCard';
+import SingleArticle from './SingleArticle';
+import { useSelector, useDispatch } from 'react-redux';
 import { Card } from 'semantic-ui-react';
 
 const DisplayArticlesData = () => {
-  const [articlesData, setArticlesData] = useState([]);
+  const dispatch = useDispatch();
+  const articles = useSelector((state) => state.articlesIndex);
+  const singleArticle = useSelector((state) => state.singleArticle);
+
   const getArticles = async () => {
-    let result = await getArticleData();
-    setArticlesData(result.data.articles);
+    const result = await getArticleData();
+    dispatch({ type: 'SET_ARTICLES_INDEX', payload: result.data.articles });
+    dispatch({ type: 'SET_SINGLE_ARTICLE' });
+  };
+
+  const fetchSingleArticle = async (event) => {
+    const id = event.target.dataset.id;
+    const response = await getSpecificArticle(id);
+    dispatch({ type: 'SET_SINGLE_ARTICLE', payload: response.data.article });
```

Comment:
If you are setting the response from this function in application state, you don't need to pass it as props but you can execute it on the level where it is being triggered, in the article card

Reply:
We fixed this one.